daemon/logger/syslog/syslog.go

```diff
@@ -0,0 +1,61 @@
+package syslog
+
+import (
+	"log/syslog"
+	"os"
+	"path"
+	"sync"
+
+	"github.com/docker/docker/daemon/logger"
+)
+
+type Syslog struct {
+	writer *syslog.Writer
+	tag    string
+	mu     sync.Mutex
+}
+
+func New(tag string) (logger.Logger, error) {
+	return &Syslog{
+		writer: nil,
+		tag:    tag,
+	}, nil
+}
+
+func (s *Syslog) Log(msg *logger.Message) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if s.writer == nil {
+		log, err := syslog.New(syslog.LOG_USER, path.Base(os.Args[0]))
+		if err != nil {
+			return err
+		}
+		s.writer = log
+	}
+	if msg.Source == "stderr" {
+		if err := s.writer.Err(s.tag + ": " + string(msg.Line)); err != nil {
+			s.writer.Close()
+			s.writer = nil
+			return err
+		}
+
+	} else {
+		if err := s.writer.Info(s.tag + ": " + string(msg.Line)); err != nil {
```

Comment:
Same sprintf here.  You can also format this message before the if block since the format is the same in both "stderr" and the else